drizzlepac/helpers/utils.py

```diff
@@ -0,0 +1,245 @@
+                                        fits.HDUList([b[kw] for kw in keys]),
+                                        ignore_keywords=ignore_keywords,
+                                        rtol=rtol,
+                                        **kwargs)
+
+    if no_assert:
+        return result
+
+    assert result.identical, result.report()
+
+
+def cmp_gen_hdrkeywords(fitsobj, base, keywords, limit=0, start=0):
+    """Generate list of FITS header elements to compare
+
+    TODO: Use correct terminology
+
+    Parameters
+    ----------
+    fitsobj: HDUList
+        TODO
+
+    base: str
+        Primary keyword
+
+    keywords: list
+        Additional keywords to use
+
+    limit: int
+        Number of extensions
+        Note: 1-indexed
+
+    start: int
+        Start with extension number
+
+    Returns
+    -------
+    output: list
+        Keywords to compare
+
+    """
+    assert isinstance(fitsobj, fits.HDUList)
+    assert isinstance(base, str)
+    assert isinstance(keywords, list)
+
+    output = list(fitsobj[base])
+
+    if limit and not start:
+        start += 1
+
+    for idx in range(start, limit + 1):
+        for key in keywords:
+            if not limit:
+                output.append(fitsobj[key])
+            else:
+                output.append(fitsobj[key, idx])
+
+    return output
+
+
+# Check strings based on words using length precision
+def word_precision_check(str1, str2, length=5):
+    """Check to strings word-by-word based for word length
+
+    The strings are checked word for word, but only for the first
+    `length` characters
+
+    Parameters
+    ----------
+    str1, str2: str
+        The strings to compare
+
+    length: int
+        The number of characters in each word to check.
+
+    Returns
+    -------
+    match: boolean
+        True if the strings match
+    """
+    words1 = str1.split()
+    words2 = str2.split()
+    if len(words1) != len(words2):
+        return False
+    for w1, w2 in zip(words1, words2):
+        if w1[:length] != w2[:length]:
+            break
+    else:
+        return True
+    return False
+
+
+def test_word_precision_check():
+    """Test word_precision_check"""
+    s1 = "a b c"
+    s2 = "aa bb cc"
+    s3 = "aa bb cc dd"
+    s4 = "aazz bbzz cczz"
+
+    assert word_precision_check(s1, s1)
+    assert not word_precision_check(s1, s2)
+    assert word_precision_check(s1, s2, length=1)
+    assert not word_precision_check(s2, s3)
+    assert word_precision_check(s2, s4, length=2)
+
+
+def abspath(filepath):
+    """Get the absolute file path"""
+    return os.path.abspath(os.path.expanduser(os.path.expandvars(filepath)))
+
+
+def download(url, dest):
+    """Simple http/https downloader
+    """
+    dest = os.path.abspath(dest)
+
+    with requests.get(url, stream=True) as r:
+        with open(dest, 'w+b') as data:
+            for chunk in r.iter_content(chunk_size=0x4000):
```

Comment:
Is chunk size a magic number?

Reply:
`16384` :grin:

Reply:
What does that mean in plain English? Should be documented. :wink:

Reply:
All this does is iterate over 16 kilobytes at a time instead of reading the entire file into memory at once. That's all.